File: SDK_BIOTIME_4_5_5N_BIOFACE/SDK_BIOTIME_4_5_5N_BIOFACE/SDK_BIOTIME_4_5_5N_BIOFACE/BIOTIME4-5/VB.NET/VB.NET_BioTimeSample/frmScreenSaver.vb
Public Class frmScreenSaver

    Public Const SLEEP_MSG_LEN As Integer = 128 * 2 'Sleep Message can be 128 characters of UNICODE
    Public Const COMPANY_NAME_LEN As Integer = 64 * 2 ' Company Name can be 64 characters of UNICODE
    Public Const CUSTOMER_NAME_LEN As Integer = 64 * 2 ' Customer Name can be 64 characters of UNICODE

    Private Sub cmdGetGlyphSize_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdGetGlyphSize.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer = 0
        Dim strXML As String

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, False)

        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        strXML = MakeXMLCommandHeader("GetSleepMsgGlyphSize")
        bRet = frmMain.BioTime.GeneralOperationXML(strXML)

        If bRet Then
            lblMessage.Text = "GetSleepMsgGlyphSize OK"
            txtGlyphWidth.Text = frmMain.BioTime.XML_ParseInt(strXML, "Width")
            txtGlyphHeight.Text = frmMain.BioTime.XML_ParseInt(strXML, "Height")
        Else
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
        End If

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, True)
    End Sub

    Private Sub frmScreenSaver_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        frmMain.Visible = True
    End Sub

    Private Sub cmdExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdExit.Click
        Close()
    End Sub

    Private Sub cmdGetCustomerInfo_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdGetCustomerInfo.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer = 0
        Dim strXML As String
        Dim strCompanyName As String = "", strCustomerName As String = ""

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, False)

        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        strXML = MakeXMLCommandHeader("GetCustomerInfo")

        bRet = frmMain.BioTime.GeneralOperationXML(strXML)

        If bRet Then
            lblMessage.Text = "GetCustomerInfo OK"
            frmMain.BioTime.XML_ParseBinaryUnicode(strXML, "CompanyName", strCompanyName, COMPANY_NAME_LEN)
            frmMain.BioTime.XML_ParseBinaryUnicode(strXML, "CustomerName", strCustomerName, CUSTOMER_NAME_LEN)

            txtCompanyName.Text = strCompanyName
            txtCustomerName.Text = strCustomerName
        Else
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
        End If

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, True)
    End Sub

    Private Sub cmdSetCustomerInfo_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdSetCustomerInfo.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer = 0
        Dim strXML As String

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, False)

        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        strXML = MakeXMLCommandHeader("SetCustomerInfo")
        frmMain.BioTime.XML_AddBinaryUnicode(strXML, "CompanyName", txtCompanyName.Text)
        frmMain.BioTime.XML_AddBinaryUnicode(strXML, "CustomerName", txtCustomerName.Text)

        bRet = frmMain.BioTime.GeneralOperationXML(strXML)

        If bRet Then
            lblMessage.Text = "SetCustomerInfo OK"
        Else
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
        End If

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, True)
    End Sub

    Private Sub cmdGetSleepMsg_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdGetSleepMsg.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer = 0
        Dim strXML As String
        Dim strSleepMessage As String = ""

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, False)

        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        strXML = MakeXMLCommandHeader("GetSleepMessage")

        bRet = frmMain.BioTime.GeneralOperationXML(strXML)

        If bRet Then
            lblMessage.Text = "GetSleepMessage OK"
            frmMain.BioTime.XML_ParseBinaryUnicode(strXML, "SleepMessage", strSleepMessage, SLEEP_MSG_LEN)

            txtSleepMessage.Text = strSleepMessage
        Else
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
        End If

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, True)
    End Sub

    Private Sub cmdSetSleepMsg_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdSetSleepMsg.Click
        Dim bRet As Boolean
        Dim vErrorCode As Integer = 0
        Dim strXML As String
        Dim strFontXML As String = ""

        Dim glyphWidth As Integer, glyphHeight As Integer
        Dim fontHeight As Integer, fontWidth As Integer, fontWeight As Integer

        lblMessage.Text = "Working..."
        Application.DoEvents()

        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, False)

        If Not bRet Then
            lblMessage.Text = gstrNoDevice
            Exit Sub
        End If

        strXML = MakeXMLCommandHeader("GetSleepMsgGlyphSize")

        bRet = frmMain.BioTime.GeneralOperationXML(strXML)
        If Not bRet Then
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
            GoTo _lExit
        End If

        glyphWidth = frmMain.BioTime.XML_ParseInt(strXML, "Width")
        glyphHeight = frmMain.BioTime.XML_ParseInt(strXML, "Height")
        fontHeight = Val(txtFontHeight.Text)
        fontWidth = Val(txtFontWidth.Text)
        fontWeight = Val(txtFontWeight.Text)


        frmMain.BioTime.XML_AddString(strFontXML, "FaceName", "Arial")
        frmMain.BioTime.XML_AddInt(strFontXML, "Height", fontHeight)
        frmMain.BioTime.XML_AddInt(strFontXML, "Width", fontWidth)
        frmMain.BioTime.XML_AddInt(strFontXML, "Weight", fontWeight)
        frmMain.BioTime.XML_AddBoolean(strFontXML, "Italic", chkItalic.Checked)
        frmMain.BioTime.XML_AddBoolean(strFontXML, "Underline", chkUnderline.Checked)
        frmMain.BioTime.XML_AddBoolean(strFontXML, "StrikeOut", chkStrikeOut.Checked)

        If chkDebugOut.Checked And txtDebugOutFile.Text <> "" Then
            frmMain.BioTime.XML_AddString(strFontXML, "DebugOut", txtDebugOutFile.Text)
        End If

        strXML = MakeXMLCommandHeader("SetSleepMessage")
        frmMain.BioTime.XML_AddBinaryGlyph(strXML, txtSleepMessage.Text, glyphWidth, glyphHeight, strFontXML)

        bRet = frmMain.BioTime.GeneralOperationXML(strXML)

        If bRet Then
            lblMessage.Text = "SetSleepMessage OK"
        Else
            frmMain.BioTime.GetLastError(vErrorCode)
            lblMessage.Text = ErrorPrint(vErrorCode)
        End If

_lExit:
        bRet = frmMain.BioTime.EnableDevice(frmMain.gMachineNumber, True)
    End Sub
End Class

File: SDK_BIOTIME_4_5_5N_BIOFACE/SDK_BIOTIME_4_5_5N_BIOFACE/SDK_BIOTIME_4_5_5N_BIOFACE/BIOTIME4-5/C#/BioTimeSampleCSharp/frmScreenSaver.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
//using System.Linq;
using System.Text;
using System.Windows.Forms;

using System.Runtime.InteropServices;

namespace SBXPCSampleCSharp
{
    public partial class frmScreenSaver : Form
    {
        public frmScreenSaver()
        {
            InitializeComponent();
        }

        AxSBXPCLib.AxSBXPC bpc;
        const int SLEEP_MSG_LEN = 128 * 2; //Sleep Message can be 128 characters of UNICODE
        const int COMPANY_NAME_LEN = 64 * 2; //Company Name can be 64 characters of UNICODE
        const int CUSTOMER_NAME_LEN = 64 * 2; //Customer Name can be 64 characters of UNICODE

        private void cmdGetCustomerInfo_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;
            string strXML  = "";
            string strCompanyName = "", strCustomerName = "";

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = bpc.EnableDevice(Program.gMachineNumber, 0);

            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;                 
            }

            util.MakeXMLRequestHeader(bpc, ref strXML, "GetCustomerInfo");

            bRet = bpc.GeneralOperationXML(ref strXML);

            if(!bRet)
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
                goto _lexit;                
            }

            lblMessage.Text = "GetCustomerInfo OK";

            bpc.XML_ParseBinaryUnicode(ref strXML, "CompanyName", ref strCompanyName, COMPANY_NAME_LEN);
            bpc.XML_ParseBinaryUnicode(ref strXML, "CustomerName", ref strCustomerName, CUSTOMER_NAME_LEN);

            txtCompanyName.Text = strCompanyName;
            txtCustomerName.Text = strCustomerName;

        _lexit:
            bRet = bpc.EnableDevice(Program.gMachineNumber, 1);
        }

        private void frmScreenSaver_Load(object sender, EventArgs e)
        {
            bpc = (AxSBXPCLib.AxSBXPC)Application.OpenForms["frmMain"].Controls["BioTime"];
        }

        private void cmdSetCustomerInfo_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;
            string strXML = "", strCompanyName = "", strCustomerName = "";

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = bpc.EnableDevice(Program.gMachineNumber, 0);

            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            strCompanyName = txtCompanyName.Text;
            strCustomerName = txtCustomerName.Text;

            util.MakeXMLRequestHeader(bpc, ref strXML, "SetCustomerInfo");
            bpc.XML_AddBinaryUnicode(ref strXML, "CompanyName", ref strCompanyName);
            bpc.XML_AddBinaryUnicode(ref strXML, "CustomerName", ref strCustomerName);

            bRet = bpc.GeneralOperationXML(ref strXML);

            if (bRet)
                lblMessage.Text = "SetCustomerInfo OK";
            else
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            bRet = bpc.EnableDevice(Program.gMachineNumber, 1);
        }

        private void frmScreenSaver_FormClosing(object sender, FormClosingEventArgs e)
        {
            Application.OpenForms["frmMain"].Visible = true;
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void cmdGetGlyphSize_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;
            string strXML = "";
            int glyphWidth, glyphHeight;

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = bpc.EnableDevice(Program.gMachineNumber, 0);

            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            util.MakeXMLRequestHeader(bpc, ref strXML, "GetSleepMsgGlyphSize");

            bRet = bpc.GeneralOperationXML(ref strXML);

            if(bRet)
            {
                lblMessage.Text = "GetSleepMsgGlyphSize OK"; 
                glyphWidth = bpc.XML_ParseInt(ref strXML, "Width");
                glyphHeight = bpc.XML_ParseInt(ref strXML, "Height");
                txtGlyphWidth.Text = Convert.ToString(glyphWidth);
                txtGlyphHeight.Text = Convert.ToString(glyphHeight);
            }
            else
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            bRet = bpc.EnableDevice(Program.gMachineNumber, 1);
        }

        private void cmdGetSleepMsg_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;
            string strXML = "";

            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = bpc.EnableDevice(Program.gMachineNumber, 0);
            
            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            util.MakeXMLRequestHeader(bpc, ref strXML, "GetSleepMessage");

            bRet = bpc.GeneralOperationXML(ref strXML);

            if(bRet)
            {
                lblMessage.Text = "GetSleepMessage OK";
                string strSleepMessage = "";
                bpc.XML_ParseBinaryUnicode(ref strXML, "SleepMessage", ref strSleepMessage, SLEEP_MSG_LEN);
                txtSleepMessage.Text = strSleepMessage;
            }
            else
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            bRet = bpc.EnableDevice(Program.gMachineNumber, 1);
        }

        private void cmdSetSleepMsg_Click(object sender, EventArgs e)
        {
            bool bRet;
            int vErrorCode = 0;
            string strXML = "", strFontXML = "", strSleepMsg;
            int glyphWidth, glyphHeight;
            int fontHeight, fontWidth, fontWeight;


            lblMessage.Text = "Working...";
            Application.DoEvents();

            bRet = bpc.EnableDevice(Program.gMachineNumber, 0);

            if(!bRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            util.MakeXMLRequestHeader(bpc, ref strXML, "GetSleepMsgGlyphSize");

            bRet = bpc.GeneralOperationXML(ref strXML);

            if(!bRet)
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
                goto _lexit;
            }

            glyphWidth = bpc.XML_ParseInt(ref strXML, "Width");
            glyphHeight = bpc.XML_ParseInt(ref strXML, "Height");

            fontHeight = Convert.ToInt32(txtFontHeight.Text == "" ? "0" : txtFontHeight.Text);
            fontWidth = Convert.ToInt32(txtFontWidth.Text == "" ? "0" : txtFontWidth.Text);
            fontWeight = Convert.ToInt32(txtFontWeight.Text == "" ? "0" : txtFontWeight.Text);
            strSleepMsg = txtSleepMessage.Text;

            bpc.XML_AddString(ref strFontXML, "FaceName", "Arial");
            bpc.XML_AddInt(ref strFontXML, "Height", fontHeight);
            bpc.XML_AddInt(ref strFontXML, "Width", fontWidth);
            bpc.XML_AddInt(ref strFontXML, "Weight", fontWeight);
            bpc.XML_AddBoolean(ref strFontXML, "Italic", chkItalic.Checked);
            bpc.XML_AddBoolean(ref strFontXML, "Underline", chkUnderline.Checked);
            bpc.XML_AddBoolean(ref strFontXML, "StrikeOut", chkStrikeOut.Checked);

            if(chkDebugOut.Checked && txtDebugOutFile.Text != "")
            {
                bpc.XML_AddString(ref strFontXML, "DebugOut", txtDebugOutFile.Text);
            }

            strXML = "";
            util.MakeXMLRequestHeader(bpc, ref strXML, "SetSleepMessage");
            bpc.XML_AddBinaryGlyph(ref strXML, strSleepMsg, glyphWidth, glyphHeight, strFontXML);

            bRet = bpc.GeneralOperationXML(ref strXML);

            if (bRet)
                lblMessage.Text = "SetSleepMessage OK";
            else
            {
                bpc.GetLastError(ref vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

        _lexit:
            bRet = bpc.EnableDevice(Program.gMachineNumber, 1);
        }

        private void cmdDebugOutFileBrowse_Click(object sender, EventArgs e)
        {
            OpenSaveDlg.ShowDialog();
            txtDebugOutFile.Text = OpenSaveDlg.FileName;
        }

    }
}
